membership-attribute-service/app/controllers/AccountController.scala

```
@@ -57,19 +57,20 @@ class AccountController extends LazyLogging {
     val maybeUserId = authenticationService.userId
     logger.info(s"Attempting to retrieve payment details for identity user: $maybeUserId")
     (for {
-      user <- EitherT(Future.successful(maybeUserId \/> "no identity cookie for user"))
-      contact <- EitherT(request.touchpoint.contactRepo.get(user).map(_ \/> s"couldn't read contact from SF for $user (TODO check the separate ERROR to find out why)"))
-      sub <- EitherT(request.touchpoint.subService.either[F, P](contact).map(_ \/> s"couldn't read sub from zuora for crmId ${contact.salesforceAccountId} (TODO check the separate WARN to find out why)"))
-      details <- EitherT(request.touchpoint.paymentService.paymentDetails(sub).map[\/[String, PaymentDetails]](\/.right))
-    } yield (contact, details).toResult).run.map {
-      case \/-(result) => {
+      user <- OptionEither.liftFutureOption(maybeUserId \/> "no identity cookie for user")
+      contact <- OptionEither(request.touchpoint.contactRepo.get(user))
+      sub <- OptionEither.liftOption(request.touchpoint.subService.either[F, P](contact).map(_ \/> s"couldn't read sub from zuora for crmId ${contact.salesforceAccountId} (TODO check the separate WARN to find out why)"))
+      details <- OptionEither.liftOption(request.touchpoint.paymentService.paymentDetails(sub).map[\/[String, PaymentDetails]](\/.right))
+    } yield (contact, details).toResult).run.run.map {
+      case \/-(Some(result)) =>
         logger.info(s"Successfully retrieved payment details result for identity user: $maybeUserId")
         result
-      }
-      case -\/(message) => {
+      case \/-(None) =>
+        logger.info(s"identity user doesn't exist in SF: $maybeUserId")
+        Ok(Json.obj())
+      case -\/(message) =>
         logger.warn(s"Unable to retrieve payment details result for identity user $maybeUserId due to $message")
         Ok(Json.obj())
```

should we return an error here?